CAP-40: Add proposal to add a ed25519 signed payload signer

leighmcculloch opened the pull request:
### What
Add a proposal to add an ed25519 signed hash signer for use alongside the precondition primitives, specifically the `extraSigners` precondition, that are proposed in CAP-21.

### Why
See within the document.

Close stellar/experimental-payment-channels#165

leighmcculloch commented on core/cap-0040.md:
```diff
@@ -0,0 +1,232 @@
+     KEY_TYPE_ED25519 = 0,
+     KEY_TYPE_PRE_AUTH_TX = 1,
+     KEY_TYPE_HASH_X = 2,
++    KEY_TYPE_ED25519_SIGNED_HASH = 4,
```
It would probably be more accurate if this was named:
```suggestion
+    KEY_TYPE_HASH_TO_SIGN_ED25519 = 4,
```

leighmcculloch replied:
Or:
```suggestion
+    KEY_TYPE_HASH_FOR_ED25519 = 4,
```

stanford-scs commented:
Hint is too expensive.  Do something without SHA.

What happened to crypto key type 3?

Be explicit about what is being signed

Instead of hash, just make it a variable-length opaque data.

Need new strkey type.  (SEP-0023)

leighmcculloch commented:
Thanks @stanford-scs. I addressed your comments. For the strkey, I'll address that after the CAP team has done an initial review of the CAP and it shows to have promise for being accepted.

stanford-scs reviewed (approved):
Very nice!
